include/seqan3/std/span

```diff
@@ -384,14 +366,23 @@ public:
     span<byte, dynamic_extent> as_writeable_bytes() const noexcept
     { return {reinterpret_cast<byte *>(data()), size_bytes()}; }
 
-    friend constexpr iterator begin(span s) noexcept { return s.begin(); }
-    friend constexpr iterator end(span s) noexcept { return s.end(); }
-
 private:
     pointer    data_;
     index_type size_;
 };
 
+template <typename span_tp, ptrdiff_t extend>
+constexpr auto begin(span<span_tp, extend> s) noexcept
+{
+    return s.begin();
+}
+
+template <typename span_tp, ptrdiff_t extend>
+constexpr auto end(span<span_tp, extend> s) noexcept
+{
+    return s.end();
+}
```

Comment:
these shouldnt be nessary, there are friends.

Reply:
They are replacing the friends on purpose. With the friends everything breaks because of infinite recursive instantiation when evaluating std::ranges::begin :unamused: